internal/handler/handler.go

```diff
@@ -0,0 +1,38 @@
+package handler
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"github.com/seventv/common/utils"
+	"github.com/seventv/compactdisc/internal/global"
+	"go.uber.org/zap"
+)
+
+func Register(gctx global.Context, session *discordgo.Session) {
+	session.AddHandler(messageCreate(gctx))
+	session.AddHandler(guildMemberAdd(gctx))
+}
+
+// gives the user the default role if they dont already have it
+// this is for users already in the guild who do not yet have it
+func messageCreate(gctx global.Context) func(s *discordgo.Session, m *discordgo.MessageCreate) {
+	return func(s *discordgo.Session, m *discordgo.MessageCreate) {
+		if !utils.Contains(m.Member.Roles, gctx.Config().Discord.DefaultRoleId) {
```

Comment:
Looks better now, but one possible issue I see is this: if `DefaultRoleID` isn't set in the config, it would be an empty string, and this would call the Discord API on every message with a failing request.

Reply:
To fix that simply just check `DefaultRoleId != ""`